### DISCUSSION TITLE
ESA CCI SM assimilation happening at sub-daily scales (05,06,07,08 times every day)

### DISCUSSION THREAD

**Discussion by Gayathridevi123** [NASA-LIS/LISF#1237] [Q&A]
Hi LIS community,

I'm trying to assimilate ESA CCI combined SM (version 6.1) in NOAH 3.9 LSM using the EnKF DA algorithm and the log file showing 

LIS cycle time: 01/01/2015 05:00:00
[INFO] Assimilating Observations using EnKF for DA instance           1
[INFO] Finished assimilating Observations using EnKF

Like these assimilating observations using EnKF for LIS cycle time 06:00:00, 07:00:00, and 08:00:00. and innovation, increment files are created 05,06,07,08 times every day. But I came to know it is not supposed to run DA at sub-daily scales (05,06,07,08 times every day). And the innovation files are showing -9999 values. 
Please find the attached config file, log file, and DA_INPUT.
[files.zip](https://github.com/NASA-LIS/LISF/files/9989782/files.zip)

I'm using LISF public 7.4 version. Can anyone give me suggestions to solve this?

Thank you
Gayathri

**Comment by Gayathridevi123** [comment]
@karsenau @bmcandr @dmocko Can you please help me with this issue?

**Comment by dmocko** [comment]
Hi @Gayathridevi123 

"bmcandr" has left the LIS team, and won't be able to help.

I have asked the DA members of the LIS team for their input to your issue.

**Reply by Gayathridevi123** [reply]
Thanks a lot @dmocko for your reply. Eagerly waiting for the DA members response.

**Comment by LIS-navari** [comment]
Hi @Gayathridevi123 

I have not used this data set before. I believe, the LIS ESACCI_sm reader assigns a time to each observation. Therefore, when soil moisture observations become available at 05, 06, 07, and 08Z, LIS assimilates observations. 

Please see 
https://github.com/NASA-LIS/LISF/blob/23a60bca4bf118306fbb2f046b50f75eb1f05cc7/lis/dataassim/obs/ESACCI_sm/read_ESACCIsm.F90#L99
and 
https://github.com/NASA-LIS/LISF/blob/23a60bca4bf118306fbb2f046b50f75eb1f05cc7/lis/dataassim/obs/ESACCI_sm/read_ESACCIsm.F90#L137

Best 
Mahdi

**Reply by Gayathridevi123** [reply]
Hi @LIS-navari 

Thank you for your response. I wonder how LIS assimilate ESA CCI data at 5,6,7,8Z time as the esa cci is daily observation file. 
I tried with the esa cci passive data by using lisf master branch. In that case also assimilation happening at these times. 
I would like to know any DA members have tried this testcase that uses ESACCI passive data (https://github.com/NASA-LIS/LISF/tree/master/lis/testcases/dataassim/ESA-CCI_passive_sm). If anyone tried kindly let me know about the assimilation time.

Thank you for your time.
Gayathri

**Comment by LIS-navari** [comment]
Hi @Gayathridevi123 @dmocko 

This is my understanding. 
Your domain is very large and covers 4-5 time zones. Therefore, soil moisture data assimilation in the far left of the domain occurs in 04 Z, and assimilation on the far right happens in 08 Z.  

Best
Mahdi

**Reply by Gayathridevi123** [reply]
Hi @LIS-navari @dmocko 

Thanks again for helping with this.. My domain is covering the whole Indian region specifically domain coordinates are

Map projection of the LIS domain:      latlon
Run domain lower left lat:          6.375
Run domain lower left lon:          66.375
Run domain upper right lat:         38.375
Run domain upper right lon:         99.875
Run domain resolution (dx):          0.25
Run domain resolution (dy):          0.25

And in the code mentioned by you looks like it should read data at 0Z and assimilate at that time. But just to confirm I will definitely try using a smaller domain and will see how it assimilates. I'll update about this.

Meanwhile I again request if any DA members tried the data assimilation testcase using esacci passive data please let me know how the assimilation time worked for that case. Thank you very much for your support. 

Gayathri

**Comment by LIS-navari** [comment]
Hi @Gayathridevi123  @dmocko 

Could you please elaborate on this statement? 
"And in the code mentioned by you looks like it should read data at 0Z and assimilate at that time. "

Please review these lines of code again 
     ESACCI_sm_struc(n)%smobs  = LIS_rc%udef
     do r=1,LIS_rc%obs_lnr(k)
        do c=1,LIS_rc%obs_lnc(k)
           grid_index = LIS_obs_domain(n,k)%gindex(c,r)
           if(grid_index.ne.-1) then 
              if(smobs(c+(r-1)*LIS_rc%obs_lnc(k)).gt.0) then             
                 ESACCI_sm_struc(n)%smobs(c,r) = &
                      smobs(c+(r-1)*LIS_rc%obs_lnc(k))                 
                 lon = LIS_obs_domain(n,k)%lon(c+(r-1)*LIS_rc%obs_lnc(k))
                 
                 lhour = 12.0
                 call LIS_localtime2gmt (gmt,lon,lhour,zone)
                 ESACCI_sm_struc(n)%smtime(c,r) = gmt
              endif
           endif
        enddo
     enddo
  endif

LIS date-time is in gmt. 

Thanks
Mahdi

**Reply by Gayathridevi123** [reply]
Hi @LIS-navari @dmocko 

My apologies for that wrong statement. I will try with a smaller domain and update you soon. 

Thank you
Gayathri

**Reply by Gayathridevi123** [reply]
Hi @LIS-navari @dmocko 

Please accept my apologies for the delayed response. I tried with a smaller domain so that the domain box is inside India and hence has only one-time zone. Now assimilation is happening at 06 and 07 times every day. 

[INFO] LIS cycle time: 01/01/2015 05:45:00
[INFO] LIS cycle time: 01/01/2015 06:00:00
 [INFO] Assimilating Observations using EnKF for DA instance           1
 [INFO] Finished assimilating Observations using EnKF
[INFO] LIS cycle time: 01/01/2015 06:15:00
 [INFO] Getting new time2 data
 [INFO] Reading GDAS file2 (I) 
 ./input/MET_FORCING/GDAS//201501/2015010106.gdas1.sfluxgrbf03.sg
 [INFO] Reading GDAS file2 (A) 
 ./input/MET_FORCING/GDAS//201501/2015010106.gdas1.sfluxgrbf03.sg
[INFO] LIS cycle time: 01/01/2015 06:30:00
[INFO] LIS cycle time: 01/01/2015 06:45:00
[INFO] LIS cycle time: 01/01/2015 07:00:00
 [INFO] Assimilating Observations using EnKF for DA instance           1
 [INFO] Finished assimilating Observations using EnKF
[INFO] LIS cycle time: 01/01/2015 07:15:00
[INFO] LIS cycle time: 01/01/2015 07:30:00

Before it was at 05,06,07 and 08 times.

Please find the attached log file. 
[lislog.zip](https://github.com/NASA-LIS/LISF/files/10276513/lislog.zip)

@LIS-navari can you please look into this matter?

Thank you
Gayathri

**Comment by LIS-navari** [comment]
Hi @Gayathridevi123 

Your original domain was very large and covers 4-5 time zones. Therefore, soil moisture data assimilation in the far left of the domain occurs in 04 Z, and assimilation on the far right happens in 08 Z. Now you are using smaller domain but your domain covers two time zones then DA is happening at 06 and 07 every day. 
Could you please explain what is the issue? 

Best
Mahdi

**Reply by Gayathridevi123** [reply]
Hi @LIS-navari 

Thanks for reply. Now I tried with a smaller domain and domain box is inside India so there is only single time zone! And assimilation happening at 2 times - 6 and 7...

Gayathri

**Comment by LIS-navari** [comment]
Hi @Gayathridevi123 

I wish I know more. When data assimilation happens two times, that means your domain covers two time zones. Please look at the reader (lines 100-140). 
https://github.com/NASA-LIS/LISF/blob/23a60bca4bf118306fbb2f046b50f75eb1f05cc7/lis/dataassim/obs/ESACCI_sm/read_ESACCIsm.F90#L99

Best
Mahdi

**Reply by Gayathridevi123** [reply]
But I am giving a very small domain that is inside India. In India there is only one time zone. So how it can assimilate at two times?

**Reply by Gayathridevi123** [reply]
My domain details
![Capture2](https://user-images.githubusercontent.com/87421619/209379306-6bb3b58c-4eed-49b0-9331-ae1ff9875d5a.PNG)

I will share config files and log files on Monday if required. Log file I already uploaded.

**Reply by dmocko** [reply]
My only advice is to closely look at the logic of the code to determine what's happening for your domain.
